python/lsst/ts/observatory/control/maintel/mtcalsys.py

```diff
@@ -0,0 +1,756 @@
+
+        """
+        # TODO (DM-44772): implement the actual function
+        return 10.0
+
+    async def change_laser_wavelength(self, wavelength: float) -> None:
+        """Change the TunableLaser wavelength setting
+
+        Parameters
+        ----------
+        wavelength : `float`
+            wavelength of the laser in nm
+        """
+
+        task_wavelength = self.rem.tunablelaser.cmd_changeWavelength.set_start(
+            wavelength=wavelength, timeout=self.long_long_timeout
+        )
+        task_focus = self.linearstage_laser_focus.cmd_moveAbsolute.set_start(
+            distance=self.calculate_laser_focus_location(wavelength)
+        )
+
+        await asyncio.gather(task_wavelength, task_focus)
+
+    async def is_ready_for_flats(self) -> bool:
+        """Designates if the calibraiton hardware is in a state
+        to take flats.
+        """
+        # TODO (DM-44310): Implement method to check that the
+        # system is ready for flats.
+        return True
+
+    async def setup_calsys(self, sequence_name: str) -> None:
+        """Setup the calibration system.
+
+        If monochromatic flats, check that laser can be enabled,
+        check temperature, and turn on laser to warm up.
+        Move linearstage_select to correct location for
+        Mono or Whitelight flats. If Monochromatic flats
+        make sure that LED stage is in the correct place.
+
+        Parameters
+        ----------
+        sequence_name : `str`
+            name of the type of configuration you will run, which is saved
+            in the configuration.yaml files
+
+        """
+        config_data = self.get_calibration_configuration(sequence_name)
+
+        calibration_type = getattr(CalibrationType, str(config_data["calib_type"]))
+
+        if calibration_type == CalibrationType.WhiteLight:
+            await self.linearstage_projector_select.cmd_moveAbsolute.set_start(
+                distance=self.ls_select_led_location, timeout=self.long_timeout
+            )
+        else:
+            await self.linearstage_led_select.cmd_moveAbsolute.set_start(
+                distance=self.led_rest_position, timeout=self.long_timeout
+            )
+            await self.linearstage_projector_select.cmd_moveAbsolute.set_start(
+                distance=self.ls_select_laser_location, timeout=self.long_timeout
+            )
+            await self.setup_laser(config_data["laser_mode"])
+            await self.rem.tunablelaser.cmd_startPropagating.start(
+                timeout=self.long_long_timeout
+            )
+
+    async def setup_laser(self, mode: str) -> None:
+        """Perform all steps for preparing the laser for monochromatic flats.
+        This includes confirming that the thermal system is
+        turned on and set at the right temperature. It also checks
+        the interlockState to confirm it's ready to propagate.
+
+        Parameters
+        ----------
+        mode : `str`
```

Comment:
This should be an enumeration no?

Reply:
Glad you pointed this out. There are two different enums: one for the XML and one in the TunableLaser. I have to use the one from the XML, right?

Reply:
yes